extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/user/RemoteAuthenticatedUser.java

```diff
@@ -67,6 +69,21 @@
      */
     private static final Pattern X_FORWARDED_FOR = Pattern.compile("^" + IP_ADDRESS_REGEX + "(, " + IP_ADDRESS_REGEX + ")*$");
 
+    /**
+     * Arbitrary attributes associated with this RemoteAuthenticatedUser object.
+     */
+    private Map<String, String> attributes = new HashMap<String, String>();
```

Comment:
I'm not sure if it's better to put this here, in this abstract class, or if this (along with the `@Override`s below) should only be done on the implementing classes, like `ModeledAuthenticatedUser`?  @mike-jumper any guidance?

Reply:
The problem with the `RemoteAuthenticatedUser` is it implements AuthenticatedUser.

Reply:
and currently `AuthenticatedUser` extends `Attributes`

Reply:
Yes, but `RemoteAuthenticatedUser` is abstract, so the `getAttributes()` and `setAttributes()` methods do not have to be implemented in it - they can be, as you've done, here, but they can also be implemented by all of the implementing classes.  Obviously implementing them in `RemoteAuthenticatedUser` requires fewer lines of code, as then you don't have to implement in both `SharedAuthenticatedUser` and `ModeledAuthenticatedUser`, I'm just not sure how "clean" this is, or if the `RemoteAuthenticatedUser` class is really the right place for implementing these attribute handlers.

Reply:
Oh yes you are right. I didn't notice this. I am going to try to set the attributes in `ModeledAuthenticatedUser`